code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -574,21 +574,19 @@ What are the archived variables for?
 		var/energy_released = 0
 		var/old_heat_capacity = heat_capacity()
 		var/burned_fuel = 0
+		burned_fuel = (1 - (N2O_DECOMPOSITION_COEFFICIENT_A  / ((private_temperature + N2O_DECOMPOSITION_COEFFICIENT_C) ** 2))) * private_sleeping_agent
```

Comment:
Add a min() to this to be certain that it won't consume more sleeping agent than exists.

Reply:
The formula can only become greater than 1 or negative if the constants are changed. is this truly necessary?

Reply:
Unless we need the performance, it's better to make our guarantees explicit. We never want to burn more fuel than actually exists, so we should make it obvious that that's guaranteed.

Reply:
done